### PR TITLE
Build Warnings Cleanup

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Description

Mainly removing unused variable warnings

## Related Issue

closes #5257 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added my user impacting change to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document. 

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/apps/makeflat/main.cpp [hunk 1]
```diff
@@ -391,10 +390,6 @@ void IsisMain() {
 
   cubeInitialized = false;
   for(currImage = 0; currImage < inList.size(); currImage++) {
-    if(Excluded(currImage)) {
-      excludedCnt ++;
-      continue;
```
Comment (AustinSanders) [inline isis/src/base/apps/makeflat/main.cpp:396]: We probably want to leave the
```cpp
if(Excluded(currImage){
  continue;
}

File: isis/src/base/objs/Pipeline/Pipeline.cpp [hunk 2]
```diff
@@ -600,11 +600,9 @@ namespace Isis {
     }
 
     //Check for non nulls instead of size ie. find last non null or use this
-    int appsSize = 0;
     QString pauseAppId = "";
 
     for (int iapp = 0; iapp < (int) p_apps.size(); iapp++)
```
Comment (AustinSanders) [inline isis/src/base/objs/Pipeline/Pipeline.cpp]: I think we can drop this empty for loop

File: isis/src/control/apps/autoseed/autoseed.cpp [hunk 3]
```diff
@@ -258,11 +256,6 @@ namespace Isis {
     for (int ov = 0; ov < overlaps.Size(); ++ov) {
       progress.CheckStatus();
 
-      if (overlaps[ov]->Size() == 1) {
```
Comment (AustinSanders) [inline isis/src/control/apps/autoseed/autoseed.cpp:261]: I think we want to keep this "if ... continue" as well

File: isis/src/control/apps/autoseed/autoseed.cpp [hunk 4]
```diff
@@ -320,12 +313,6 @@ namespace Isis {
         continue;
       }
 
-      // No points were seeded in this polygon, so collect some stats and move on
-      if (points.size() == 0) {
-        stats_tolerance++;
-        continue;
```
Comment (AustinSanders) [inline isis/src/control/apps/autoseed/autoseed.cpp:326]: Same here

**Review by AustinSanders — CHANGES_REQUESTED** [review]

File: isis/src/base/apps/makeflat/main.cpp (hunk 1 above)
Comment (amystamile-usgs) [reply isis/src/base/apps/makeflat/main.cpp:396]: Why's that? Wouldn't the loop continue regardless?

File: isis/src/base/objs/Pipeline/Pipeline.cpp (hunk 2 above)
Comment (amystamile-usgs) [reply isis/src/base/objs/Pipeline/Pipeline.cpp]: 👍

File: isis/src/base/apps/makeflat/main.cpp (hunk 1 above)
Comment (AustinSanders) [reply isis/src/base/apps/makeflat/main.cpp:396]: `continue` inside a for-loop will skip the remaining logic in the for-loop and move to the next iteration.  If Excluded(currImage) is true, we want to short-circuit the rest of the logic in the loop and move to the next image.

Comment (amystamile-usgs) [reply isis/src/base/apps/makeflat/main.cpp:396]: Oh gotcha. That makes sense.

**Comment by amystamile-usgs** [comment]
Looking into test failures:
hinoise_app_test_binningIsOne (Failed) 
hinoise_app_test_binningNotOne (Failed)

**Comment by AustinSanders** [comment]
It looks like one more test is failing:

isis_unit_test_Pipeline

**Comment by amystamile-usgs** [comment]
> It looks like one more test is failing:
> 
> isis_unit_test_Pipeline

Tests should be fixed now.

**Review by AustinSanders — APPROVED** [review]

**Pull request merged**